examples/deploy-krops.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script demonstrates how to setup a nix-bitcoin node with krops.
+# The node is deployed to a minimal NixOS QEMU VM.
+# Running this script leaves no traces on your host system.
+
+# This demo is a template for your own experiments.
+# Run with option `--interactive` or `-i` to start a shell for interacting with
+# the node.
+
+# MAKE SURE TO REPLACE the SSH identity file if you use this script for
+# anything serious.
+
+if [[ ! -v IN_NIX_SHELL ]]; then
+    echo "Running script in nix shell env..."
+    cd "${BASH_SOURCE[0]%/*}"
+    exec nix-shell --run "./${BASH_SOURCE[0]##*/} $*"
+fi
+
+source qemu-vm/run-vm.sh
+
+echo "Building the target VM"
+# Build the initial VM to which the nix-bitcoin node is deployed to via krops
+nix-build --out-link $tmpDir/vm - <<'EOF'
+(import <nixpkgs/nixos> {
+  configuration = { lib, ... }: {
+    imports = [ <qemu-vm/vm-config.nix>  ];
+    services.openssh.enable = true;
+
+    # Silence the following warning that appears when deploying via krops:
+    # warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
+    nix.nixPath = lib.mkForce [];
+  };
+}).vm
+EOF
+
+vmNumCPUs=4
+vmMemoryMiB=2048
+sshPort=60734
+# Start the VM in the background
+runVM $tmpDir/vm $vmNumCPUs $vmMemoryMiB $sshPort
+
+# Build the krops deploy script
+export sshPort
+nix-build --out-link $tmpDir/krops-deploy - <<'EOF'
+let
+  krops = (import <nix-bitcoin> {}).krops;
+
+  extraSources = {
+    # Skip uploading nixpkgs to the target node.
+    # This works because /nix/store is shared with the target VM.
+    nixpkgs.symlink = toString <nixpkgs>;
+
+    nixos-config.file = toString <krops-vm-configuration.nix>;
+    "krops-configuration.nix".file = toString ./krops-configuration.nix;
+
+    qemu-vm.file = toString <qemu-vm>;
+  };
+in
+krops.pkgs.krops.writeCommand "krops-deploy" {
+  source = import <krops/sources.nix> { inherit extraSources krops; };
+  force = true;
+  target = {
+    user = "root";
+    host = "127.0.0.1";
+    port = builtins.getEnv "sshPort";
```

Comment:
Because of the surrounding quoting with `'EOF'` instead of `EOF`.
`'EOF'` is more robust because `"` and `$` don't have to be escaped, but it doesn't support variable expansion.

Reply:
I see... didn't even know about `'EOF'`

Reply:
Sorry, of course, `"` doesn't have to be escaped in normal `EOF` either. But `\` must still be escaped inside `'EOF'`.